### PR TITLE
Changelog 1.66

### PR DESCRIPTION
**Pull request by xFrednet**
It's really nice to see a changelog with so many suggestion fixes and improvements. Not much else to say. This should be merged with the coming release on 2022-12-15. For the reviewer, please review it and approve it if it looks good. The merge should wait until the release :) 

---

changelog: none

### CODE REVIEW

**Comment by xFrednet** [comment]
On days like these, I'm happy to have an alias for fixups ^^. Now everything should be green :)

**Comment by xFrednet** [comment]
Okay, now everything is green, it just wanted to check my alias for `push --force-with-lease`

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,12 +6,181 @@ document.
 
 ## Unreleased / Beta / In Rust Nightly
 
-[b52fb523...master](https://github.com/rust-lang/rust-clippy/compare/b52fb523...master)
+[4f142aa1...master](https://github.com/rust-lang/rust-clippy/compare/4f142aa1...master)
 
-## Rust 1.65
+## Rust 1.66
 
 Current stable, released 2022-11-03
```
Comment (giraffate) [inline CHANGELOG.md:13]:
```suggestion
Current stable, released 2022-12-15
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -6,11 +6,180 @@ document.
...
+* [`use_self`]: No longer lints in proc macros
+  [#9454](https://github.com/rust-lang/rust-clippy/pull/9454)
+* [`never_loop`]: Now takes `let ... else` statements into consideration.
+  [#9496](https://github.com/rust-lang/rust-clippy/pull/9496)
+* [`default_numeric_fallback`]: Now ignores constants
+  [#9636](https://github.com/rust-lang/rust-clippy/pull/9636)
+* [`uninit_vec`]: No longer lints `Vec::set_len(0)`
+  [#9519](https://github.com/rust-lang/rust-clippy/pull/9519)
+* [`arithmetic_side_effects`]: Now ignores references
+  [9507](https://github.com/rust-lang/rust-clippy/pull/9507)
```
Comment (Alexendoo) [inline CHANGELOG.md:114]:
```suggestion
* [`arithmetic_side_effects`]: Now ignores references to integer types
  [#9507](https://github.com/rust-lang/rust-clippy/pull/9507)
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -6,11 +6,180 @@ document.
...
+
+### ICE Fixes
+
+* [`unnecessary_to_owned`]: Avoid ICEs in favor of false negatives if information is missing
+  [#9505](https://github.com/rust-lang/rust-clippy/pull/9505)
+* [`manual_range_contains`]: No longer ICEs on values behind references
+  [#9627](https://github.com/rust-lang/rust-clippy/pull/9627)
+* [`needless_pass_by_value`]: No longer ICEs on unsized `dyn Fn` arguments
+  [#9531](https://github.com/rust-lang/rust-clippy/pull/9531)
+* `*_interior_mutable_const` lints: no longer ICE on const unions containing `!Freeze` types
```
Comment (Alexendoo) [inline CHANGELOG.md:173]:
```suggestion
* `*_interior_mutable_const` lints: no longer ICE on const unions containing `!Freeze` types
  [#9539](https://github.com/rust-lang/rust-clippy/pull/9539)
```

**Review by Alexendoo — COMMENTED** [review]
LGTM with these two nits

**Review by Alexendoo — APPROVED** [review]

**Review by giraffate — APPROVED** [review]

**Pull request merged**